odb-compiler/include/odbc/util/Plugin.hpp

```diff
@@ -16,7 +16,7 @@ class Plugin
 
     static Plugin* load(const char* filename);
 
-    bool loadKeywords(KeywordDB* db) const;
+    bool loadKeywords(KeywordDB& db) const;
```

Comment:
I prefer output params to always be pointers, because it's clear at the call site. ```loadKeywords(db);``` isn't clear if keywords are being loaded from the db to the plugin, or vice versa, but ```loadKeywords(&db);``` is.

Reply:
Fair enough. The way I have seen it, is that I prefer to use references when they cannot be null, so the type system enforces it. However, I see your point as well, so will change it back.

Reply:
Yeah the no-null enforcement argument is true too. I agree in every other situation except for out parameters